src/Core/Managed/Shared/Extensibility/AutocollectedMetricsExtractor.cs

```diff
@@ -252,6 +263,29 @@ private void ExtractMetrics(ITelemetry fromItem)
             }
         }
 
+        private bool EnsureItemNotSampled(ITelemetry item)
+        {
+            ISupportSampling potentiallySampledItem = item as ISupportSampling;
+
+            if (potentiallySampledItem != null && potentiallySampledItem.SamplingPercentage.HasValue)
```

Comment:
Ok. However, is your use case what we recommend? This makes the condition complex. For the above use case we should instead expose a special boolean like "SkipSampling" or similar.

Also, the sampling processor instead uses (100.0 - 1.0E-12) as the threshold.